### ISSUE TITLE
Tidy pivot calibration and add RANSAC version

### ISSUE THREAD

**Issue by thompson318** [SciKit-Surgery/scikit-surgerycore#25]
In GitLab by @MattClarkson on Mar 8, 2020, 12:27

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 8, 2020, 16:39

I was also going to add spherical sphere fitting, but i just realised this requires Levenberg Marquart. What about adding scipy to scikit-surgerycore? 

scikit-surgerycore was meant to be basic algorithms. Does that include non-linear optimisation?

@StephenThompson @ThomasDowrick ?

Is it kind of comparable to scikit-surgeryimage requiring opencv?
It is a transitive problem though, as all things that include core, will end up with scipy. 

The reason I ask, is that otherwise, you end up putting fairly simple algorithms (e.g, linear algebra + non-lin optimisation), in another library?

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 8, 2020, 16:43

mentioned in commit 28171f5ac6c1036523346f29e55936f9833eac25

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 8, 2020, 16:43

mentioned in commit f99a1d663edeca72b0a911d1371f52e5f84c36c4

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 9, 2020, 13:45

mentioned in commit 76310a9f4db7f9f0d15d04309c654016a0a5dcb0

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 9, 2020, 14:00

changed title from **{-Add sphere fitting for pivot calibrat-}ion** to **{+Tidy pivot calibration and add RANSAC vers+}ion**

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 9, 2020, 14:05

mentioned in commit b33f43a7241ec7ac05222528122ecf2bcca38eda

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 9, 2020, 14:13

mentioned in commit fce661e6b428600dbae008c22170a721d5bb12d9

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Mar 9, 2020, 14:32

mentioned in commit e9e1fcd19b3e1d3eeb990d236f52037a865061a0

**Comment by thompson318** [comment]
In GitLab by @StephenThompson on Mar 10, 2020, 11:50

I don't think adding scipy to core is a good idea, reasons below.
I did a brief survey of what we've got that depends on scipy and surgery-core. We've got a lot of libraries and applications that use core but not scipy (utils, davinci, snappysonic, bard, all the trackers etc). So we'd be adding a lot of redundant dependency to all of these. 
I'd suggest creating surgery-calibration and potentially surgery-camera-calibration to hold these algorithms. We should take full advantages of pip's ability to makes managing multiple small dependencies easy.
